### PR TITLE
feat: cli: lotus info cmd

### PR DESCRIPTION
**Pull request by rjan90**
## Related Issues
Implements most of the metrics listed in #5766.

## Proposed Changes
Add a `lotus info`  command that prints info about the lotus node:

```
Network: butterflynet
Chain: [sync ok] [basefee 100 aFIL] [epoch 194446]
Peers to: [publish messages 13] [publish blocks 13]
Chain health: 68% [unhealthy]

Default address: 
      t1p34z3ex72fa3v7h7thkwl6n3ggqzgne3sz5ls3y [22.869 FIL]

Wallet: 12 addresses
      Total balance: 748.537 FIL
      Market locked: 166.638 μFIL
      Market available: 1.5 FIL

Payment Channels: 2 channels

Deals: 6, 762 MiB
      StorageDealActive: 5 deals, 635 MiB
      StorageDealSealing: 1 deals, 127 MiB

Bandwidth:
      TotalIn  TotalOut  RateIn  RateOut
      4.9 MB   4.2 MB    0 B/s   0 B/s
```

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [x] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] CI is green

### CODE REVIEW

File: cli/info.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+	switch {
+	case health > 85:
+		fmt.Printf("Chain health: %.f%% [%s]\n", health, color.GreenString("healthy"))
+	case health < 85:
+		fmt.Printf("Chain health: %.f%% [%s]\n", health, color.RedString("unhealthy"))
+	}
+
+	fmt.Println()
+
+	addr, err := fullapi.WalletDefaultAddress(ctx)
```
Comment (magik6k) [inline cli/info.go:89]: What happens if there is no default set?

File: cli/info.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	}
+
+	switch {
+	case time.Now().Unix()-int64(head.MinTimestamp()) < int64(build.BlockDelaySecs*3/2): // within 1.5 epochs
+		fmt.Printf("[%s]", color.GreenString("sync ok"))
+	case time.Now().Unix()-int64(head.MinTimestamp()) < int64(build.BlockDelaySecs*5): // within 5 epochs
+		fmt.Printf("[%s]", color.YellowString("sync slow (%s behind)", time.Now().Sub(time.Unix(int64(head.MinTimestamp()), 0)).Truncate(time.Second)))
+	default:
+		fmt.Printf("[%s]", color.RedString("sync behind! (%s behind)", time.Now().Sub(time.Unix(int64(head.MinTimestamp()), 0)).Truncate(time.Second)))
+	}
```
Comment (magik6k) [inline cli/info.go:55]: If this is duplicated with the code in lotus-miner info, we should put this in a function here, and call that from lotus-miner to not duplicate logic.

File: cli/info.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+		return err
+	}
+
+	fmt.Printf(" [epoch %s]\n", color.MagentaString(("%d"), status.SyncStatus.Epoch))
+	fmt.Printf("Peers to: [publish messages %d] [publish blocks %d]\n", status.PeerStatus.PeersToPublishMsgs, status.PeerStatus.PeersToPublishBlocks)
+
+	//Chain health calculated as percentage: amount of blocks in last finality / very healthy amount of blocks in a finality (900 epochs * 5 blocks per tipset)
+	health := (100 * (900 * status.ChainStatus.BlocksPerTipsetLastFinality) / (900 * 5))
+	switch {
+	case health > 85:
```
Comment (magik6k) [inline cli/info.go:81]: For setting the threshold - what % do we see on mainnet?

**Review by magik6k — COMMENTED** [review]
This is really cool!

File: cli/info.go (hunk 1 above)
Comment (rjan90) [reply cli/info.go:69]: Fixed up the logic to only print the default address and balance if err == nil: https://github.com/filecoin-project/lotus/pull/9233/commits/8a45e8c8af836d917759dd1a44701a5361a325ec

File: cli/info.go (hunk 2 above)
Comment (rjan90) [reply cli/info.go]: Its duplicated with the code in `lotus-miner info` so I put it in a function here: https://github.com/filecoin-project/lotus/pull/9233/commits/818153a1e71a14f913572f1b5659fb8ad250b4af

Was able to call it on the `lotus-miner info` side as well but for that I needed to `GetFullNodeAPIV1` on that side. I might be missing some understanding between v0api and v1api, so might bug you a bit during a standup session.

File: cli/info.go (hunk 3 above)
Comment (rjan90) [reply cli/info.go:61]: Its been hovering around 4.8 - 4.9 blocks on average in the last finality on mainnet while I have been monitoring it, so around 96% - 98%

Could maybe try to get some historical avarages as well for setting the threshold more realistic.

**Review by magik6k — COMMENTED** [review]
`make gen` is sad

**Review by magik6k — APPROVED** [review]

**Pull request merged**